Sources/Clappr_iOS/Classes/Base/FullScreen/FullScreenStateHandler.swift

```diff
@@ -1,48 +1,37 @@
 protocol FullscreenStateHandler {
 
     var core: Core { get }
-    var isOnFullscreen: Bool { get }
 
     init(core: Core)
 
     func set(fullscreen: Bool)
-    func enterInFullscreen(_: EventUserInfo)
     func enterInFullscreen()
-
-    func exitFullscreen(_: EventUserInfo)
     func exitFullscreen()
 }
 
-extension FullscreenStateHandler {
-
-    func enterInFullscreen() {
-        enterInFullscreen([:])
-    }
-
-    func exitFullscreen() {
-        exitFullscreen([:])
-    }
-
-    var isOnFullscreen: Bool {
-        return core.mediaControl?.fullscreen ?? false
-    }
-}
-
 struct FullscreenByApp: FullscreenStateHandler {
 
     var core: Core
 
     func set(fullscreen: Bool) {
-        core.mediaControl?.fullscreen = fullscreen
+        guard core.isFullscreen != fullscreen else { return }
+        core.isFullscreen = fullscreen
+        if fullscreen {
+            core.trigger(InternalEvent.willEnterFullscreen.rawValue)
+            core.trigger(InternalEvent.didEnterFullscreen.rawValue)
```

Comment:
I'm having an issue with those events: when the `didEnterFullscreen` is triggered the view is in the middle of the process of going to fullscreen. This should be triggered only after the player is *actually* in fullscreen.

What do you guys think?